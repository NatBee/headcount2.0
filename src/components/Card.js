import React from 'react';
import '../styles/Card.css'

const Card = (props) => {

const cardDisplay = Object.keys(props['data'].data).map( year => <li> {year} : {props['data'].data[year]} </li>)

  return (
    <article className='card'>
      <p>{props['data'].location}</p>
      <ul>{cardDisplay}</ul>
      
    </article>
    ) 
}

export default Card