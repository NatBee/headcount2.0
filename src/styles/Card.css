.card {
  border: 1px solid;
  width: 250px;
  display: inline-block;
  margin: 0 5px 10px 0;
  /*height: 120px;*/
  vertical-align: top;
  padding: 5px;
}

.card li {
  list-style-type: none;  
}