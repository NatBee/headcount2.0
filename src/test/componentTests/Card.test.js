import React from 'react';
import ReactDOM from 'react-dom';
import Card from '../../components/Card';
import { shallow } from 'enzyme';
const mockProps = {
  data: {
    data:{  
      2004:1,
      2005:1,
      2006:1,
      2007:1,
      2008:1,
      2009:1,
      2010:1,
      2011:1,
      2012:1,
      2013:1,
      2014:1
    },
    location: 'COLORADO'}  
  };

const mockProps2 = {
  data: {
    data:{  
      2004:0,
      2005:0,
      2006:0,
      2007:0,
      2008:0,
      2009:0,
      2010:0,
      2011:0,
      2012:0,
      2013:0,
      2014:0
    },
    location: 'COLORADO'}  
}  

it('should match the snapshot', () => {
  const renderedComponent = shallow(<Card {...mockProps} />)
  expect(renderedComponent).toMatchSnapshot()
})

it('should not have a className of .lower for data < 0.5', () => {
  const renderedComponent = shallow(<Card {...mockProps} />);
  expect(renderedComponent.find('.lower').length).toEqual(0);
})

it('should not have a className of .higher for data >= 0.5', () => {
  //change mock data for this case to have all below .5

  const renderedComponent = shallow(<Card {...mockProps2}/>);
  expect(renderedComponent.find('.higher').length).toEqual(0);
})  