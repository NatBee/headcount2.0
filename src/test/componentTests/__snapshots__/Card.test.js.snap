// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
        Object {
            "data": Object {
              "2004": 1,
              "2005": 1,
              "2006": 1,
              "2007": 1,
              "2008": 1,
              "2009": 1,
              "2010": 1,
              "2011": 1,
              "2012": 1,
              "2013": 1,
              "2014": 1,
            },
            "location": "COLORADO",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          COLORADO
</h3>,
        <ul>
          <li
                    className="higher"
          >
                     
                    2004
                     : 
                    1
                     
          </li>
          <li
                    className="higher"
          >
                     
                    2005
                     : 
                    1
                     
          </li>
          <li
                    className="higher"
          >
                     
                    2006
                     : 
                    1
                     
          </li>
          <li
                    className="higher"
          >
                     
                    2007
                     : 
                    1
                     
          </li>
          <li
                    className="higher"
          >
                     
                    2008
                     : 
                    1
                     
          </li>
          <li
                    className="higher"
          >
                     
                    2009
                     : 
                    1
                     
          </li>
          <li
                    className="higher"
          >
                     
                    2010
                     : 
                    1
                     
          </li>
          <li
                    className="higher"
          >
                     
                    2011
                     : 
                    1
                     
          </li>
          <li
                    className="higher"
          >
                     
                    2012
                     : 
                    1
                     
          </li>
          <li
                    className="higher"
          >
                     
                    2013
                     : 
                    1
                     
          </li>
          <li
                    className="higher"
          >
                     
                    2014
                     : 
                    1
                     
          </li>
</ul>,
      ],
      "className": "card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO",
        },
        "ref": null,
        "rendered": "COLORADO",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="higher"
>
               
              2004
               : 
              1
               
</li>,
            <li
              className="higher"
>
               
              2005
               : 
              1
               
</li>,
            <li
              className="higher"
>
               
              2006
               : 
              1
               
</li>,
            <li
              className="higher"
>
               
              2007
               : 
              1
               
</li>,
            <li
              className="higher"
>
               
              2008
               : 
              1
               
</li>,
            <li
              className="higher"
>
               
              2009
               : 
              1
               
</li>,
            <li
              className="higher"
>
               
              2010
               : 
              1
               
</li>,
            <li
              className="higher"
>
               
              2011
               : 
              1
               
</li>,
            <li
              className="higher"
>
               
              2012
               : 
              1
               
</li>,
            <li
              className="higher"
>
               
              2013
               : 
              1
               
</li>,
            <li
              className="higher"
>
               
              2014
               : 
              1
               
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "2004",
                " : ",
                1,
                " ",
              ],
              "className": "higher",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "2004",
              " : ",
              1,
              " ",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "2005",
                " : ",
                1,
                " ",
              ],
              "className": "higher",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "2005",
              " : ",
              1,
              " ",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "2006",
                " : ",
                1,
                " ",
              ],
              "className": "higher",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "2006",
              " : ",
              1,
              " ",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "2007",
                " : ",
                1,
                " ",
              ],
              "className": "higher",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "2007",
              " : ",
              1,
              " ",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "2008",
                " : ",
                1,
                " ",
              ],
              "className": "higher",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "2008",
              " : ",
              1,
              " ",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "2009",
                " : ",
                1,
                " ",
              ],
              "className": "higher",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "2009",
              " : ",
              1,
              " ",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "6",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "2010",
                " : ",
                1,
                " ",
              ],
              "className": "higher",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "2010",
              " : ",
              1,
              " ",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "7",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "2011",
                " : ",
                1,
                " ",
              ],
              "className": "higher",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "2011",
              " : ",
              1,
              " ",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "8",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "2012",
                " : ",
                1,
                " ",
              ],
              "className": "higher",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "2012",
              " : ",
              1,
              " ",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "9",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "2013",
                " : ",
                1,
                " ",
              ],
              "className": "higher",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "2013",
              " : ",
              1,
              " ",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "10",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "2014",
                " : ",
                1,
                " ",
              ],
              "className": "higher",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "2014",
              " : ",
              1,
              " ",
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            COLORADO
</h3>,
          <ul>
            <li
                        className="higher"
            >
                         
                        2004
                         : 
                        1
                         
            </li>
            <li
                        className="higher"
            >
                         
                        2005
                         : 
                        1
                         
            </li>
            <li
                        className="higher"
            >
                         
                        2006
                         : 
                        1
                         
            </li>
            <li
                        className="higher"
            >
                         
                        2007
                         : 
                        1
                         
            </li>
            <li
                        className="higher"
            >
                         
                        2008
                         : 
                        1
                         
            </li>
            <li
                        className="higher"
            >
                         
                        2009
                         : 
                        1
                         
            </li>
            <li
                        className="higher"
            >
                         
                        2010
                         : 
                        1
                         
            </li>
            <li
                        className="higher"
            >
                         
                        2011
                         : 
                        1
                         
            </li>
            <li
                        className="higher"
            >
                         
                        2012
                         : 
                        1
                         
            </li>
            <li
                        className="higher"
            >
                         
                        2013
                         : 
                        1
                         
            </li>
            <li
                        className="higher"
            >
                         
                        2014
                         : 
                        1
                         
            </li>
</ul>,
        ],
        "className": "card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO",
          },
          "ref": null,
          "rendered": "COLORADO",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="higher"
>
                 
                2004
                 : 
                1
                 
</li>,
              <li
                className="higher"
>
                 
                2005
                 : 
                1
                 
</li>,
              <li
                className="higher"
>
                 
                2006
                 : 
                1
                 
</li>,
              <li
                className="higher"
>
                 
                2007
                 : 
                1
                 
</li>,
              <li
                className="higher"
>
                 
                2008
                 : 
                1
                 
</li>,
              <li
                className="higher"
>
                 
                2009
                 : 
                1
                 
</li>,
              <li
                className="higher"
>
                 
                2010
                 : 
                1
                 
</li>,
              <li
                className="higher"
>
                 
                2011
                 : 
                1
                 
</li>,
              <li
                className="higher"
>
                 
                2012
                 : 
                1
                 
</li>,
              <li
                className="higher"
>
                 
                2013
                 : 
                1
                 
</li>,
              <li
                className="higher"
>
                 
                2014
                 : 
                1
                 
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "2004",
                  " : ",
                  1,
                  " ",
                ],
                "className": "higher",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "2004",
                " : ",
                1,
                " ",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "2005",
                  " : ",
                  1,
                  " ",
                ],
                "className": "higher",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "2005",
                " : ",
                1,
                " ",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "2006",
                  " : ",
                  1,
                  " ",
                ],
                "className": "higher",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "2006",
                " : ",
                1,
                " ",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "2007",
                  " : ",
                  1,
                  " ",
                ],
                "className": "higher",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "2007",
                " : ",
                1,
                " ",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "2008",
                  " : ",
                  1,
                  " ",
                ],
                "className": "higher",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "2008",
                " : ",
                1,
                " ",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "5",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "2009",
                  " : ",
                  1,
                  " ",
                ],
                "className": "higher",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "2009",
                " : ",
                1,
                " ",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "6",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "2010",
                  " : ",
                  1,
                  " ",
                ],
                "className": "higher",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "2010",
                " : ",
                1,
                " ",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "7",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "2011",
                  " : ",
                  1,
                  " ",
                ],
                "className": "higher",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "2011",
                " : ",
                1,
                " ",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "8",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "2012",
                  " : ",
                  1,
                  " ",
                ],
                "className": "higher",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "2012",
                " : ",
                1,
                " ",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "9",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "2013",
                  " : ",
                  1,
                  " ",
                ],
                "className": "higher",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "2013",
                " : ",
                1,
                " ",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "10",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "2014",
                  " : ",
                  1,
                  " ",
                ],
                "className": "higher",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "2014",
                " : ",
                1,
                " ",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
